.clang-format

```diff
@@ -0,0 +1,2 @@
+#not sure what to put in here, everything else just formatted like this for now
+BasedOnStyle: GNU
```

Comment:
GNU's probably chill, i might fuck around with the other ones incase there's something even more based

Reply:
think we probably want 4 spaces for indentation, but we could also do like tabs for indents  spaces for alignment since then you can just have your text editor display whatever tab width you like

Reply:
🤓